pkg/services/sqlstore/transactions.go

```diff
@@ -14,53 +15,18 @@ func (ss *SqlStore) InTransaction(ctx context.Context, fn func(ctx context.Conte
 }
 
 func (ss *SqlStore) inTransactionWithRetry(ctx context.Context, fn func(ctx context.Context) error, retry int) error {
-	sess, err := startSession(ctx, ss.engine, true)
-	if err != nil {
-		return err
-	}
-
-	defer sess.Close()
-
-	withValue := context.WithValue(ctx, ContextSessionName, sess)
-
-	err = fn(withValue)
-
-	// special handling of database locked errors for sqlite, then we can retry 3 times
-	if sqlError, ok := err.(sqlite3.Error); ok && retry < 5 {
-		if sqlError.Code == sqlite3.ErrLocked {
-			sess.Rollback()
-			time.Sleep(time.Millisecond * time.Duration(10))
-			ss.log.Info("Database table locked, sleeping then retrying", "retry", retry)
-			return ss.inTransactionWithRetry(ctx, fn, retry+1)
-		}
-	}
-
-	if err != nil {
-		sess.Rollback()
-		return err
-	}
-
-	if err = sess.Commit(); err != nil {
```

Comment:
Removed here but the same code is in `inTransactionWithRetryCtx`. Not sure if I am missing something but it seems like this prevents us having nested transaction blocks reusing session. I would expect this to commit only if I created the session (so I am outer most transaction block) but not if I am reusing session from the parent (which mean the session does not have to be finished).

Did not test this so it is possible I am reading this wrong. It does not error out as `sess.Commit()` is noop if the transaction is already committed.

cc @bergquist